### PR TITLE
Build script and readme fixes

### PR DESCRIPTION
**Pull request by SenRamakri**
Fixes for build.sh failing with tags -debug not defined issue and minor updates to readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ Different subsystems have different test suites. Running these may require root
 Example, from `$GOPATH/src/github.com/armPelionEdge/maestro`:
 
 ```
-DEBUG=1 DEBUG2=1 ./build.sh && cd networking && sudo \
+DEBUG=1 DEBUG2=1 ./build.sh && cd networking && sudo -E \
```
Comment (SenRamakri) [inline README.md]: We need -E here so that the environment($GOROOT, $GOBIN, $GOPATH) is preserved when using sudo as well.

File: build.sh [hunk 2]
```diff
@@ -53,10 +53,19 @@ sed -i -e "s/BUILD_DATE/${DATE}/g" maestroutils/status.go
 # highlight errors: https://serverfault.com/questions/59262/bash-print-stderr-in-red-color
 # color()(set -o pipefail;"$@" 2>&1>&3|sed $'s,.*,\e[31m&\e[m,'>&2)3>&1
 
+#Developers may have a GOBIN defined. If not define it to current dir.
+if [ -z "$GOBIN" ]
```
Comment (cmonr) [inline build.sh]: A monir nit since the pattern already exists in this file.

```suggestion
if [ -z "$GOBIN" ]; then
```

File: build.sh [hunk 3]
```diff
@@ -53,10 +53,19 @@ sed -i -e "s/BUILD_DATE/${DATE}/g" maestroutils/status.go
...
+  GOBIN=./bin
+fi
+
 if [[ "${1:-}" != "preprocess_only" ]]; then
-	if [[ -n "${TIGHT:-}" ]]; then
-	    go build "${GOTAGS}" -ldflags="-s -w" "$@" github.com/armPelionEdge/maestro/maestro 
-	else
-	    go build "${GOTAGS}" "$@" github.com/armPelionEdge/maestro/maestro 
-	fi
+  mkdir -p $GOBIN
```
Comment (cmonr) [inline build.sh]:
```suggestion
  mkdir -p "${GOBIN}"
```

Add brackets to fail if the variable is undefined, and double quotes to make sure variable is treated only as a string (ie, what would/could happen if the variable has a space).

**Comment by SenRamakri** [comment]
@cmonr - Those are good suggestions, changes are applied.

File: build.sh [hunk 4]
```diff
@@ -54,7 +54,7 @@ sed -i -e "s/BUILD_DATE/${DATE}/g" maestroutils/status.go
 # color()(set -o pipefail;"$@" 2>&1>&3|sed $'s,.*,\e[31m&\e[m,'>&2)3>&1
 
 #Developers may have a GOBIN defined. If not define it to current dir.
-if [ -z "$GOBIN" ]
+if [ -z "${GOBIN}" ]; then
 then
```
Comment (cmonr) [inline build.sh]: Sorry, GitHub sugestions only work one line at a time right now.

Meant for this line to also be removed.

**Pull request merged**